activelearning/dataset/grid.py

```diff
@@ -22,12 +23,14 @@ def __init__(
         X_data,
         y_data=None,
         normalize_scores=True,
+        state2result=None,
```

Comment:
I think it's important to add docstring and type hints here. That will be the way to be sure about the format/representations that are expected.